WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/BlogDashboardPersonalizeCardCell.swift

```diff
@@ -0,0 +1,90 @@
+import UIKit
+import SwiftUI
+
+final class BlogDashboardPersonalizeCardCell: DashboardCollectionViewCell {
+    private var blog: Blog?
+    private weak var presentingViewController: BlogDashboardViewController?
+
+    private let personalizeButton = UIButton(type: .system)
+
+    // MARK: - Initializers
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - View setup
+
+    private func setupView() {
+        let titleLabel = UILabel()
+        titleLabel.text = Strings.buttonTitle
+        titleLabel.font = WPStyleGuide.fontForTextStyle(.subheadline, fontWeight: .semibold)
+        titleLabel.adjustsFontForContentSizeCategory = true
+
+        let imageView = UIImageView(image: UIImage(named: "personalize")?.withRenderingMode(.alwaysTemplate))
+        imageView.tintColor = .label
+
+        let spacer = UIView()
+        spacer.translatesAutoresizingMaskIntoConstraints = false
+        spacer.widthAnchor.constraint(greaterThanOrEqualToConstant: 8).isActive = true
+
+        let contents = UIStackView(arrangedSubviews: [titleLabel, spacer, imageView])
+        contents.alignment = .center
+        contents.isUserInteractionEnabled = false
+
+        personalizeButton.accessibilityLabel = Strings.buttonTitle
+        personalizeButton.setBackgroundImage(.renderBackgroundImage(fill: .tertiarySystemFill), for: .normal)
+        personalizeButton.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+
+        let container = UIView()
+        container.layer.cornerRadius = 10
+        container.addSubview(personalizeButton)
+        container.addSubview(contents)
+
+        personalizeButton.translatesAutoresizingMaskIntoConstraints = false
+        container.pinSubviewToAllEdges(personalizeButton)
+
+        contents.translatesAutoresizingMaskIntoConstraints = false
+        container.pinSubviewToAllEdges(contents, insets: .init(allEdges: 16))
+
+        contentView.addSubview(container)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        contentView.pinSubviewToAllEdges(container)
+    }
```

Comment:
not a blocker: Could a `UIButton`'s text and images properties and insets simplify this by avoiding separate subviews? I'm curious about your thoughts here.

Reply:
Making `UIButton` look like this isn't easy. By default, it displays the image on the left side, and there is no (good) way to set flexible spacing for the image. I wish it was as easy to use as the SwiftUI `Button`.

There is a new `UIButton.Configuration` that almost gets you there, but it's iOS 15+ only.

<img width="971" alt="Screenshot 2023-03-24 at 9 54 47 PM" src="https://user-images.githubusercontent.com/1567433/227679875-ea8fc90c-4cd8-4dec-91cc-ef66d37a50cd.png">

There is a way to do it with negative image/title insets, but then again, there is flexible spacing. I could use `layoutSubviews` to adjust the spacing dynamically based on the container and the title size. But it would've been over-complicated.

I considered using a simple gesture recognizer for handling taps on this card, but `UIButton` has a nice highlighted state and works better for accessibility.

Reply:
Thanks for sharing! I agree about `UIButton` image/title insets being complicated.